src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,261 @@
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
+    KeyBindingViewModel::KeyBindingViewModel(const Control::KeyChord& keys, const Model::Command& cmd) :
+        _Keys{ keys },
+        _KeyChordText{ Model::KeyChordSerialization::ToString(keys) },
+        _Command{ cmd }
+    {
+        // Add a property changed handler to our own property changed event.
+        // This propagates changes from the settings model to anybody listening to our
+        //  unique view model members.
+        PropertyChanged([this](auto&&, const PropertyChangedEventArgs& args) {
+            const auto viewModelProperty{ args.PropertyName() };
+            if (viewModelProperty == L"Keys")
+            {
+                _KeyChordText = Model::KeyChordSerialization::ToString(_Keys);
+                _NotifyChanges(L"KeyChordText");
+            }
+            else if (viewModelProperty == L"IsContainerFocused" ||
+                     viewModelProperty == L"IsEditButtonFocused" ||
+                     viewModelProperty == L"IsHovered" ||
+                     viewModelProperty == L"IsAutomationPeerAttached" ||
+                     viewModelProperty == L"IsInEditMode")
+            {
+                _NotifyChanges(L"ShowEditButton");
+            }
+        });
+    }
+
+    bool KeyBindingViewModel::ShowEditButton() const noexcept
+    {
+        return (IsContainerFocused() || IsEditButtonFocused() || IsHovered() || IsAutomationPeerAttached()) && !IsInEditMode();
+    }
+
+    void KeyBindingViewModel::ToggleEditMode()
+    {
+        // toggle edit mode
+        IsInEditMode(!_IsInEditMode);
+        if (_IsInEditMode)
+        {
+            // if we're in edit mode,
+            // pre-populate the text box with the current keys
+            _ProposedKeys = KeyChordText();
+            _NotifyChanges(L"ProposedKeys");
+        }
+    }
+
+    void KeyBindingViewModel::AttemptAcceptChanges()
+    {
+        auto args{ make_self<RebindKeysEventArgs>(_Keys, _Keys) };
+        try
+        {
+            // Attempt to convert the provided key chord text
+            const auto newKeyChord{ KeyChordSerialization::FromString(_ProposedKeys) };
+            args->NewKeys(newKeyChord);
+        }
+        catch (hresult_invalid_argument)
+        {
+            // Converting the text into a key chord failed
+            // TODO CARLOS:
+            //  This is tricky. I still haven't found a way to reference the
+            //  key chord text box. It's hidden behind the data template.
+            //  Ideally, some kind of notification would alert the user, but
+            //  to make it look nice, we need it to somehow target the text box.
+        }
+        _RebindKeysRequestedHandlers(*this, *args);
+    }
+
     Actions::Actions()
     {
         InitializeComponent();
+    }
 
-        _filteredActions = winrt::single_threaded_observable_vector<Command>();
+    Automation::Peers::AutomationPeer Actions::OnCreateAutomationPeer()

```

Comment:
What exactly is the deal with this? Why does the `KeyBindingViewModel` need to know if the automation peer is attached?

Reply:
Added a comment.

This was a nice little extra thing I added. The "edit" button will _always_ be shown if a screen reader is attached. This way, the screen reader knows that there's a button there.

Reply:
oh, okay that makes sense. Good thinking!